lib/addons/keda/index.ts

```diff
@@ -0,0 +1,130 @@
+import { Construct } from 'constructs';
+import { ManagedPolicy } from "aws-cdk-lib/aws-iam";
+import { HelmAddOn, HelmAddOnUserProps, HelmAddOnProps } from "../helm-addon";
+import { ClusterInfo, Values } from "../../spi";
+import { setPath, createNamespace } from "../../utils";
+
+/**
+ * User provided options for the Helm Chart
+ */
+export interface KedaAddOnProps extends HelmAddOnUserProps {
+    /**
+     * Version of the helm chart to deploy
+     */    
+    version?: string,
+    /**
+     * Name of the KEDA operator
+     */
+    kedaOperatorName?: string,
+    /**
+     * Specifies whether a service account should be created. 
+     */
+    createServiceAccount?: boolean,
+    /**
+     * The name of the service account to use. If not set and create is true, a name is generated.
+     */
+    kedaServiceAccountName?: string,
+    /**
+     * securityContext: fsGroup
+     * Check the workaround for SQS Scalar with IRSA https://github.com/kedacore/keda/issues/837#issuecomment-789037326
+     */   
+    podSecurityContextFsGroup?: number,
+    /**
+     * securityContext:runAsUser
+     * Check the workaround for SQS Scalar with IRSA https://github.com/kedacore/keda/issues/837#issuecomment-789037326
+     */   
+    securityContextRunAsGroup?: number,
+    /**
+     * securityContext:runAsGroup
+     * Check the workaround for SQS Scalar with IRSA https://github.com/kedacore/keda/issues/837#issuecomment-789037326
+     */
+    securityContextRunAsUser?: number,
+    /**
+     * Th Dictionary of MAnaged IAM Roles which Sercice Account needs for IRSA Eg: irsaRoles: {"cloudwatch":"CloudWatchFullAccess", "sqs":"AmazonSQSFullAccess"}
+     */   
+    irsaRoles?: {}
+
+}
+
+/**
+ * Default props to be used when creating the Helm chart
+ */
+const defaultProps: HelmAddOnProps & KedaAddOnProps = {
+  name: "blueprints-keda-addon",
+  chart: "keda",
+  namespace:"keda",
+  version: "2.7.1",
+  release: "keda",
+  repository:  "https://kedacore.github.io/charts",
+  values: {},
+  createServiceAccount: true,
+  kedaOperatorName: "keda-operator",
+  kedaServiceAccountName: "keda-operator",
+  irsaRoles: {"cloudwatch":"CloudWatchFullAccess"}
```

Comment:
should it be just a `string[]`? Is there any significance of the key, like `cloudwatch` in this default?

Reply:
converted it into string[]
Fixed